### ISSUE TITLE
multi-relationship classification?

### ISSUE THREAD

**Issue by Quan-zzx** [Maelic/SGG-Benchmark#37]
Hello, I have a dataset，the same subject and object have multiple relationships，does the framework support multi relationship classification?

**Comment by Maelic** [comment]
Hello, I will say yes and no. The relation decoding is multi-label because for every possible subject-object pair, a list of probabilities is computed for each predicate. So in theory, you can select more than one predicate per subject-object pair. However, in practice during training the learning is done such that the cross-entropy loss will optimize only the top relation. To modify this, you will need to change the loss and the dataloader a bit (remove [this part](https://github.com/Maelic/SGG-Benchmark/blob/286e1c14e26e653ac51811391b81dc2cdd136d23/sgg_benchmark/data/datasets/visual_genome.py#L220) for instance).
The metric that validate multi-label relationships is called No Graph Constraint Recall and is already implemented [here](https://github.com/Maelic/SGG-Benchmark/blob/286e1c14e26e653ac51811391b81dc2cdd136d23/sgg_benchmark/data/datasets/evaluation/vg/sgg_eval.py#L491). You will have to activate it [by modifying this line](https://github.com/Maelic/SGG-Benchmark/blob/286e1c14e26e653ac51811391b81dc2cdd136d23/sgg_benchmark/data/datasets/evaluation/vg/vg_eval.py#L23).

**Comment by Quan-zzx** [comment]
> Hello, I will say yes and no. The relation decoding is multi-label because for every possible subject-object pair, a list of probabilities is computed for each predicate. So in theory, you can select more than one predicate per subject-object pair. However, in practice during training the learning is done such that the cross-entropy loss will optimize only the top relation. To modify this, you will need to change the loss and the dataloader a bit (remove [this part](https://github.com/Maelic/SGG-Benchmark/blob/286e1c14e26e653ac51811391b81dc2cdd136d23/sgg_benchmark/data/datasets/visual_genome.py#L220) for instance). The metric that validate multi-label relationships is called No Graph Constraint Recall and is already implemented [here](https://github.com/Maelic/SGG-Benchmark/blob/286e1c14e26e653ac51811391b81dc2cdd136d23/sgg_benchmark/data/datasets/evaluation/vg/sgg_eval.py#L491). You will have to activate it [by modifying this line](https://github.com/Maelic/SGG-Benchmark/blob/286e1c14e26e653ac51811391b81dc2cdd136d23/sgg_benchmark/data/datasets/evaluation/vg/vg_eval.py#L23).

ok，thanks，
I'll give it a try
